packages/tse-backend/src/modules/exchange-data/exchange-data.service.ts

```diff
@@ -0,0 +1,298 @@
+    );
+
+    for (const exchange of exchangeInstances) {
+      const exchangeInstance =
+        this.exchangeRegistryService.getExchange(exchange);
+      if (exchangeInstance && exchangeInstance.has.fetchTickers) {
+        try {
+          const tickers = await exchangeInstance.fetchTickers();
+          pairs.push(...Object.keys(tickers));
+        } catch (error) {
+          this.logger.error(
+            `Error fetching tickers from ${exchange}: ${error}`,
+          );
+        }
+      }
+    }
+    return Array.from(new Set(pairs));
+  }
+
+  async getTickerPrice(command: GetTickerPriceCommand) {
+    const { exchange, symbol } = command;
+    const upperCaseSymbol = symbol.toUpperCase();
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance || !exchangeInstance.has.fetchTicker) {
+      throw new Error(
+        'Exchange does not support fetchTicker or is not configured.',
+      );
+    }
+    this.logger.log(
+      `Fetching ticker price from ${exchangeInstance.name} for ${upperCaseSymbol}`,
+    );
+
+    try {
+      const ticker = await exchangeInstance.fetchTicker(upperCaseSymbol);
+      return {
+        pair: upperCaseSymbol,
+        price: ticker.last,
+      } as TickerPriceResponse;
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  async getMultipleTickerPrices(command: GetMultipleTickerPricesCommand) {
+    const { exchangeNames, symbols } = command;
+    const fetchPromises: Promise<TickerPriceResponse>[] = [];
+    const results: {
+      [exchange: string]: { [symbol: string]: TickerPriceResponse };
+    } = {};
+
+    const exchangeNamesArray = exchangeNames.split(',');
+    const symbolsArray = symbols.split(',');
+
+    exchangeNamesArray.forEach((exchangeName) => {
+      symbolsArray.forEach((symbol) => {
+        const tpCommand = new GetTickerPriceCommand(exchangeName, symbol);
+        const fetchPromise = this.getTickerPrice(tpCommand)
+          .then((price) => {
+            if (!results[exchangeName]) {
+              results[exchangeName] = {};
+            }
+            results[exchangeName][symbol] = price;
+            return price;
+          })
+          .catch((error) => {
+            this.logger.error(
+              `Error fetching ticker price for ${exchangeName}:${symbol}: ${error}`,
+            );
+            return null;
+          });
+        fetchPromises.push(fetchPromise);
+      });
+    });
+
+    await Promise.all(fetchPromises);
+    return results;
+  }
+
+  async getSupportedSymbols(command: GetSupportedSymbolsCommand) {
+    const { exchange } = command;
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance) {
+      throw new Error(`Exchange ${exchange} is not configured.`);
+    }
+    this.logger.log(`Fetching supported symbols from ${exchangeInstance.name}`);
+
+    try {
+      await exchangeInstance.loadMarkets();
```

Comment:
This can also probably be cached if it's an API call, if so add an //TODO comment as well

Reply:
I agree it should be cached: 73d8b9b4